src/univariate/continuous/laplace.jl

```diff
@@ -114,8 +114,27 @@ rand(rng::AbstractRNG, d::Laplace) =
 
 #### Fitting
 
-function fit_mle(::Type{<:Laplace}, x::Array)
-    xc = copy(x)
-    a = median!(xc)
-    Laplace(a, StatsBase.mad!(xc, center=a))
+function fit_mle(::Type{<:Laplace}, x::AbstractArray{T}) where {T <: Real}
+    μ = median(x)
+    return Laplace(μ, mean(abs.(x .- μ)))
```

Comment:
Probably more efficient would be

```suggestion
    return Laplace(μ, StatsBase.meanad(x, μ))
```

Reply:
But maybe it expects two vectors?

Reply:
From StatsBase.jl,

# MeanAD: mean absolute deviation
"""
    meanad(a, b)
Return the mean absolute deviation between two arrays: `mean(abs(a - b))`.
"""
meanad(a::AbstractArray{T}, b::AbstractArray{T}) where {T<:Number} =
    L1dist(a, b) / length(a)

Seems it wants both arguments to be arrays.

Reply:
But yes, this can be implemented more efficiently.  Would you prefer a simple loop + accumulator?

Reply:
I see. I guess the most efficient implementation would then be more similar to the existing one: one could reuse the array that is used for computing the median. `median` creates an array internally IIRC but one can use `median!` instead with a preallocated array `xc` (as in the current implementation) and then update it with `xc .= abs.(x .- m)` where `m` is the median. I'll add a suggestion below.